reasoner/controller/ConcludableController.java

```diff
@@ -241,22 +243,33 @@ private void mayRequestConnection(REQ conclusionRequest) {
 
         public static class Match extends Processor<Map<Variable, Concept>, ConceptMap, Match.Request, Match> {
 
-            private final Concludable concludable;
-
             Match(
                     Driver<Match> driver, Driver<ConcludableController.Match> controller, Concludable concludable,
                     Context context, ConceptMap bounds, Set<Variable.Retrievable> unboundVars,
                     Map<Conclusion, Set<Unifier>> conclusionUnifiers,
                     Supplier<FunctionalIterator<ConceptMap>> traversalSuppplier, Supplier<String> debugName
             ) {
-                super(driver, controller, context, bounds, unboundVars, conclusionUnifiers, traversalSuppplier,
+                super(driver, controller, concludable, context, bounds, unboundVars, conclusionUnifiers, traversalSuppplier,
                       debugName);
-                this.concludable = concludable;
+            }
+
+
+            @Override
+            public void setUp() {
+                // TODO: Add a find first optimisation when all variables are bound
+                if (concludable.isRelation() && concludable.generating().isPresent() && bounds.contains(concludable.generating().get().id())) {
+                    // Optimisation: If we know the relation instance, just do a lookup.
+                    setHubReactive(fanInFanOut(this));
+                    mayAddTraversal(true);
+                } else {
```

Comment:
we can generalise this optimisation:
```
                if (concludable.isInferredAnswer(bounds)) {
                    // don't trigger rules ever
                    if (!bounds are fully binding concludable) {
                        mayAddTraversal(true);
                    } else {
                        return singleton bounds as answer
                    }
                } else {
                    mayAddTraversal(false);
                    triggerRules();
                }
```

you'll need to add `Concludable.isAnswer` which returns true if the concept map provided is an answer that fits as either an inferred or non-inferred answer